server/resp/src/test/java/org/infinispan/server/resp/StringCommandsTest.java

```diff
@@ -551,6 +552,16 @@ public void testGetset() {
       assertThat(redis.get("key")).isEqualTo("another");
    }
 
+   @Test
+   public void testGetsetWrongType() {
+      RedisCommands<String, String> redis = redisConnection.sync();
+      redis.lpush("key","value");
+      try {
+      redis.getset("key", "shouldfail");
+      } catch (Exception e) {}
+      assertWrongType(() -> {} , () -> redis.get("key"));
```

Comment:
Just some nitpick to the test. To validate both the exception thrown and that the entry is still a list:

```suggestion
      assertWrongType(() -> redis.lpush("key","value"), () -> redis.getset("key", "shouldfail"));
      assertWrongType(() -> {} , () -> redis.get("key"));
      assertThat(redis.lrange("key", 0, -1)).containsExactly("value");
```